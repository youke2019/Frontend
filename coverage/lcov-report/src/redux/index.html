<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> src/redux
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.05% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/84</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/25</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>36/72</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="69.7" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 69%;"></div><div class="cover-empty" style="width:31%;"></div></div></td>
	<td data-value="69.7" class="pct medium">69.7%</td>
	<td data-value="33" class="abs medium">23/33</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="9.09" class="pct low">9.09%</td>
	<td data-value="11" class="abs low">1/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="22" class="abs high">22/22</td>
	</tr>

<tr>
	<td class="file low" data-value="reducers.js"><a href="reducers.js.html">reducers.js</a></td>
	<td data-value="24.49" class="pic low"><div class="chart"><div class="cover-fill" style="width: 24%;"></div><div class="cover-empty" style="width:76%;"></div></div></td>
	<td data-value="24.49" class="pct low">24.49%</td>
	<td data-value="49" class="abs low">12/49</td>
	<td data-value="36" class="pct low">36%</td>
	<td data-value="25" class="abs low">9/25</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="48" class="abs low">12/48</td>
	</tr>

<tr>
	<td class="file high" data-value="state.js"><a href="state.js.html">state.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Aug 01 2019 01:16:28 GMT+0800 (GMT+08:00)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
<script src="..\..\block-navigation.js"></script>
</body>
</html>
