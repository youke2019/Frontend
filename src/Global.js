import Storage from 'react-native-storage'
import AsyncStorage from '@react-native-community/async-storage'

let storage = new Storage({
    size: 1000,
    storageBackend: AsyncStorage,
    defaultExpires: 1000 * 60 * 60 * 24, //1 day
    enableCache: true,
    sync: {}
})

global.storage = storage
global.baseUrl = 'http://192.168.42.146:8080'