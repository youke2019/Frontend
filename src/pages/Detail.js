import React from "react";
import {View, Text, FlatList, Alert,StyleSheet,ScrollView} from "react-native";
import axios from "axios";
import { EmitError, HandleError } from '../utils/ErrorAlert'

export const Title = (props) =>(
  <View style = {styles.Title}>
    <Text> {props.title} </Text>
    <Text> {props.content}</Text>
    {/*父子组件间传值*/}
  </View>
)


const Time = (props) =>(
  <View style = {styles.Title}>
    <Text> 上课时间 :</Text>
    <Text> {props.bw} 至 {props.ew} 周 {props.eo}周 周{props.date} 第 {props.bs}至{props.es} 节</Text>
    {/*父子组件间传值*/}
  </View>
)
const dateMap =["","一","二","三","四","五","六","日"];
const weekMap ={
  o: "单周",
  e: "双周",
  b: "全周"
}
class CourseInfo extends React.Component{

    render(){
        const {course} = this.props;
        console.log(course)
        return(
            <View style = { styles.container}>

                <Title title = {"课程"} content = { course.course_name}/>
                <View style ={styles.box}>
                    <Title title ={ "课程号"} content = {course.course_id}/>
                    <Title title ={"学分" } content = {course.course_credits}/>
                    <Title title = {"是否为通识"} content ={course.general? "是":"否"}/>
                    <Title title = {"通识类型"} content = {course.general_type}/>
                    <Title title = {"选课备注"} content = {""}/>
                    <Text> 教学班信息</Text>
                    <FlatList
                        data={course.classes}
                        keyExtractor={(item) => item.classname.toString()}
                        renderItem={({item}) =>
                            <View style={{height:200}}>
                                <Title title={"教学班"} content={item.classname}/>
                                <Title title={"任课老师"} content={item.teacher_name}/>
                                <FlatList data={item.classSegments}
                                          renderItem={({item}) =>
                                              <View>
                                                  <Title title={"上课地点"} content={item.courseroom}/>
                                                  <Text>上课时间</Text>
                                                  <Time bw = {item.begin_week} ew = {item.end_week} eo = {dateMap[item.odd_or_even]} date = {dateMap[item.week]} bs = {item.begin_sec} es = {item.end_sec}/>
                                              </View>
                                          }
                                          keyExtractor={(item) => item.class_sec_id.toString()}
                                />
                            </View>
                        }
                    />
                </View>

            </View>
        )
    }
}
/*
Class CommentTitle extends React.Component{
    render(){
        return(
            <View style = {styles.Title}>
                <Text> 评论 <Text>
                <Text> {this.props.tag}</Text>
            </View>
        )
    }
}
Class QuestionTitle extends React.Component{
    render(){
        return(
            <View style = {styles.Title}>
                <Text> 问答区 <Text>
            </View>
        )
    }
}
Class EvaluateTitle extends React.Component{
    render(){
        return(
            <View style = {styles.Title}>
                <Text> 评测 <Text>
            </View>
        )
    }
}
*/
class Detail extends React.Component {
    state = {
        courseInfo :"",
    }

    static navigationOptions = ({navigation}) => ({
        title : 'details for course ${navigation.state.params.course_id}'
    })
    //标题

    setCourse = (data) =>{
        this.setState({
            courseInfo:data
        })
    }

    //what's beneath is the function that will execute once the component is created
    componentWillMount() {
        const params = this.props.navigation.state.params;
        axios.get(baseUrl + '/courses/specific', {
          params:params
        }).then((response) => {
            this.setCourse(response.data)
        }).catch(error => {
            console.log(error)
            EmitError({ error_msg:"获取课程信息时发生了错误" })
        })
    }

    render() {
        return (
            <View>
                <CourseInfo course ={this.state.courseInfo} />
            </View>
        )
    }
}

export default Detail


const styles = StyleSheet.create({
  Title: {
    backgroundColor: '#F5FCFF',
  },
  box:{
    borderStyle: 'dashed',
    borderWidth : 1,
  },
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    top : 50,
    bottom: 50,
  },
});
