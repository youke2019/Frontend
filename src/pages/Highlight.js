import React from 'react'
import { Image, FlatList, Text, View, ScrollView, TouchableOpacity, StyleSheet } from 'react-native'
import { connect } from 'react-redux'
import { Divider } from 'react-native-elements'
import HighlightCard from '../components/HighlightCard'
import { getAllHighlight, getNumberHighlight } from '../utils/DataRequest'

class Highlight extends React.Component {
  state = {
    highlights: [],
    receivedAll: false,
    is_refreshing:false,
  }
  newHighlight = () => {
    this.props.navigation.navigate('NewHighlight', {
      user_id: this.props.user_info.id,
      callBack: this.refresh
    })
  }
  componentDidMount () {
    this.getMoreData(20)
  }

  refresh = () => {
    this.refreshData()
  }
  getMoreData = (size)=>{
    const {highlights,receivedAll} = this.state
    if(receivedAll) return;
    const currNum = highlights.length
    this.setState(({
      is_refreshing: true
    }))
    getNumberHighlight(currNum, size, this.props.user_info.id)
      .then(response =>{
        this.setState({
          highlights: highlights.concat(response.data),
          is_refreshing:false,
        })
        if(response.data.length < size)
          this.setState({receivedAll:true})
      })
      .catch(err =>{
        console.log(err)
        this.setState({ is_refreshing: false })
      })
  }
  getTenMoreData=()=>{
    this.getMoreData(10);
  }

  refreshData = () => {
    const {highlights} = this.state
    const currNum = highlights.length
    this.setState({
      is_refreshing: true
    })
    getNumberHighlight(0,currNum,this.props.user_info.id)
      .then((response) => {
        console.log(response)
        this.setState({
          highlights: response.data,
          is_refreshing: false,
          receivedAll:false,
        })}) //刷新后，允许新的数据加入
      .catch(err => {
        this.setState({ is_refreshing: false })
        console.log(err)
      })
  }
  _renderItem=({ item }) =>
    <View style={{ flexDirection: 'column' }}>
      <HighlightCard data={item} user_id={this.props.user_info.id} refresh={this.refresh}/>
      <Divider style={{ height: 5, backgroundColor: 'whitesmoke' }}/>
    </View>
  _keyExtractor=(item) => item.video_id.toString()
  render () {
    const { highlights, is_refreshing } = this.state
    console.log(highlights)
    return (
      <View style={styles.base_container}>
        <View style={styles.header}>
          <View style={styles.header_center}>
            <Image source={{ uri: 'planet' }} style={styles.header_center_img}/>
            <Text style={styles.header_text}>
              精彩瞬间
            </Text>
          </View>
          <TouchableOpacity
            style={styles.top_right_touchable}
            onPress={this.newHighlight}
          >
            <Image source={{ uri: 'picture_yellow' }} style={styles.camera}/>
          </TouchableOpacity>
        </View>
        <Divider style={{ backgroundColor: 'black' }}/>
        <FlatList
          style={{ height: '100%', width: '100%' }}
          keyboardShouldPersistTaps={'handled'}
          refreshing={is_refreshing}
          onEndReached={this.getTenMoreData}
          onEndReachedThreshold={0.8}
          onRefresh={this.refresh}
          data={highlights}
          renderItem={this._renderItem}
          keyExtractor={this._keyExtractor}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  base_container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderTopWidth: 35,
    borderTopColor: 'whitesmoke'
  },
  header: {
    width: '100%',
    height: 50,
    flexDirection: 'row',
    backgroundColor: 'whitesmoke',
    alignItems: 'center',
    justifyContent: 'flex-end'
  },
  header_text: {
    color: '#000000',
    fontSize: 25,
    fontFamily: '字魂95号-手刻宋'
  },
  header_center: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  header_center_img: {
    height: 35,
    width: 35,
    marginRight: 10
  },
  camera: {
    height: 35,
    width: 35
  },
  top_right_touchable: {
    marginLeft: 80,
    marginRight: 20
  },
  divider: {
    backgroundColor: 'blue'
  }
})
const mapStateToProps = (state) => ({
  user_info: state.user_info
})

export default connect(
  mapStateToProps
)(Highlight)