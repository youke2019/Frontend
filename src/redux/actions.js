export const UPDATE_USER_INFO = 'UPDATEUSERINFO'
export const LOAD_USER_INFO = 'LOADUSERINFO'

export const updateUserInfo = (data) => ({
    type:UPDATE_USER_INFO,
    data
})

export const loadUserInfo = (data) => ({
    type:LOAD_USER_INFO,
    data
})

