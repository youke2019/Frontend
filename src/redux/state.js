const initialState = {
    user_info: null,
    course_list: null,
    login_ready:false,
}

export default initialState