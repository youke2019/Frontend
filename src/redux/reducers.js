import { combineReducers } from 'redux'
import {
    UPDATE_USER_INFO,
    LOAD_USER_INFO,
    CLEAR_USER_INFO,
    SEARCH_COURSES,
} from "./actions.js"
import initialState from "./state";
import axios from 'axios'


function userInfoReducer(state=initialState.user_info, action) {
    switch (action.type) {
        case UPDATE_USER_INFO:
            storage.save({
                key: 'user',
                data: action.data
            })
            state = action.data
            return state
        case LOAD_USER_INFO:
            state = action.data
            return state
        case CLEAR_USER_INFO:
            state = null
            return state
        default:
            return state
    }
}

function courseReducer(state=initialState.course_list, action) {
    switch (action.type) {
        case SEARCH_COURSES:
            state = action.data
            return state
        default:
            return state
    }
}

export const combinedReducer = combineReducers({
    user_info : userInfoReducer,
    course_list: courseReducer,
})