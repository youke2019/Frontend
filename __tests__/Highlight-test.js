import renderer from 'react-test-renderer';
import Detail from '../src/pages/Detail'
import Global from '../src/Global'
import { createStore } from 'redux'
import { combinedReducer } from '../src/redux/reducers'
import { Provider } from 'react-redux'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16/build'
import Highlight from '../src/pages/Highlight'
import React from 'react'

const initialState = {
  user_info: {
    id: "30673CFC-4D4C-4FC1-9FE4-000B3D9DAA9A",
  },
  course_list: null,
  login_ready:false,
}
const store = createStore(combinedReducer,initialState)
configure({adapter: new Adapter()})

const navigationMock = {
  navigate: jest.fn(()=>{})
}

jest.mock("../src/utils/DataRequest.js")

test('renders highlight page correctly', (done) => {
  const tree = renderer.create(
    <Highlight store = {store} navigation = {navigationMock}/>
  ).toJSON();
  const tree_err = renderer.create(
    <Highlight store = {store} navigation = {navigationMock}/>
  ).toJSON();
  setTimeout(()=>{
    expect(tree).toMatchSnapshot();
    expect(tree_err).toMatchSnapshot();
    done();
  },2000)
});

describe("<Highlight /> component",()=>{
  const wrapper = shallow(
    <Highlight store = {store} navigation = {navigationMock} />
  );
  const instance = wrapper.dive().dive().instance();
  console.log(instance)
  it("can navigate to multiple pages",()=>{
      instance.newHighlight();
      instance.refresh();
  })
})