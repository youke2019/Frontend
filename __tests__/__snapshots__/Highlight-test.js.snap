// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders highlight page correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopColor": "whitesmoke",
      "borderTopWidth": 35,
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "whitesmoke",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "planet",
          }
        }
        style={
          Object {
            "height": 35,
            "marginRight": 10,
            "width": 35,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "字魂95号-手刻宋",
            "fontSize": 25,
          }
        }
      >
        精彩瞬间
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 80,
          "marginRight": 20,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "picture_yellow",
          }
        }
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "height": 0.5,
      }
    }
  />
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;

exports[`renders highlight page correctly 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopColor": "whitesmoke",
      "borderTopWidth": 35,
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "whitesmoke",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "flex-end",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "planet",
          }
        }
        style={
          Object {
            "height": 35,
            "marginRight": 10,
            "width": 35,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "字魂95号-手刻宋",
            "fontSize": 25,
          }
        }
      >
        精彩瞬间
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 80,
          "marginRight": 20,
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "picture_yellow",
          }
        }
        style={
          Object {
            "height": 35,
            "width": 35,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "height": 0.5,
      }
    }
  />
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="handled"
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={true}
      />
    }
    refreshing={true}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <RCTRefreshControl />
    <View />
  </RCTScrollView>
</View>
`;
